sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -1036,27 +1040,82 @@ public PagedFlux<ConfigurationSetting> listConfigurationSettings(SettingSelector
+                                                        httpResponse.getHeaders(),
+                                                        null,
+                                                        continuationToken,
+                                                        null));
+                                    }
+                                    return Mono.error(throwable);
+                                })
+                            .map(pagedResponse -> toConfigurationSettingWithPagedResponse(pagedResponse));
+                }),
+                nextLink -> withContext(context -> {
+                    int pageETagListSize = (matchConditionsList == null || matchConditionsList.isEmpty())
+                            ? 0
+                            : matchConditionsList.size();
+                    String nextPageETag = null;
+                    int pageETagIndexValue = pageETagIndex.get();
+                    if (pageETagIndexValue < pageETagListSize) {
+                        nextPageETag = matchConditionsList.get(pageETagIndexValue).getIfNoneMatch();
+                        pageETagIndex.set(pageETagIndexValue + 1);
+                    }
+                    return serviceClient
+                            .getKeyValuesNextSinglePageAsync(
+                                    nextLink,
+                                    acceptDateTime,
+                                    null,
+                                    nextPageETag,
+                                    addTracingNamespace(context))
+                            .onErrorResume(
+                                    HttpResponseException.class,
+                                    (Function<Throwable, Mono<PagedResponse<KeyValue>>>) throwable -> {
+                                        HttpResponseException e = (HttpResponseException) throwable;
+                                        HttpResponse httpResponse = e.getResponse();
+                                        String continuationToken = parseNextLink(httpResponse.getHeaderValue("link"));
+                                        if (httpResponse.getStatusCode() == 304) {
+                                            return Mono.just(
+                                                    new PagedResponseBase<>(
+                                                            httpResponse.getRequest(),
+                                                            httpResponse.getStatusCode(),
+                                                            httpResponse.getHeaders(),
+                                                            null,
+                                                            continuationToken,
+                                                            null));
+                                        }
+                                        return Mono.error(throwable);
+                                    })
```

Comment:
Let's extract this into it's own method as it's the same logic used above and both should be updated at the same time.